spmd/tensor/api.py

```diff
@@ -176,44 +247,22 @@ def from_local(
         if placements is None:
             placements = [Replicate() for _ in range(device_mesh.ndim)]
 
-        tensor_shape = list(local_tensor.size())
-        for idx, placement in enumerate(placements):
-            # device_list = device_mesh.mesh[idx]
-            if isinstance(placement, Shard):
-                shard_dim = placement.dim
-                # recover tensor shape on the shard dim
-                tensor_shape[shard_dim] = tensor_shape[
-                    shard_dim
-                ] * device_mesh.size(idx)
-            elif isinstance(placement, Replicate):
-                if run_check:
+        if run_check:
+            # TODO: by default check tensor metas across rank, and see if we need
+            # to make the run_check inside autograd function
+            for idx, placement in enumerate(placements):
+                if placement.is_replicate():
                     # broadcast rank 0 tensor to all ranks
                     # only broadcast if run_check is True
                     local_tensor = local_tensor.contiguous()
```

Comment:
Is this contiguous call ok?
Note that this means that you will have a `clone` in the autograd graph if the Tensor was not differentiable.

If you don't want that clone in the graph, then yes, you can move this logic to the custom Function